docs/field_types.md

```diff
@@ -312,3 +312,89 @@ $field->setOptions([
     // Your options here
 ]);
 ```
+
+Callback
+--------
+
+The Callback column aims to offer almost as much flexibility as the Twig column, but without requiring the creation of a template.
+You simply need to specify a callback, which allows you to transform the 'data' variable on the fly.
+
+By default it uses the name of the field, but you can specify the path
+alternatively. For example:
+
+<details open><summary>PHP</summary>
```

Comment:
To match existing documentation, please add YAML example as well.

Reply:
I did not include documentation for the YAML format because I believe it's impossible to define a callback field in this format.

Reply:
Feature is not complete then and cannot be merged until YAML support will be dropped.

Here is example on how callback can be implemented with non-php config format.

https://symfony.com/doc/current/reference/constraints/Callback.html#external-callbacks-and-closures

Reply:
I think we can do it. The repository can be a callback in YAML.
```yaml
app_book_by_english_authors:
    driver:
        name: doctrine/orm
        options:
            class: App\Entity\Book
            repository:
                method: [expr:service('app.english_books_query_builder'), create]
```